src/targets/VectorLogPotential.jl

```diff
@@ -0,0 +1,56 @@
+"""
+Supports user-defined log potentials that take in vectors as input.
+"""
+@concrete struct VectorLogPotential
+    """
+    User-coded function for the target log potential that assumes the state is a vector.
+    """
+    target_log_potential
```

Comment:
While technically OK to put functions into struct, a more natural way to do that is to use dispatch. But OK for now.

Reply:
Keeping VectorLogPotential as-is for now